test/parse_request.cc

```diff
@@ -1,18 +1,143 @@
+constexpr uint32_t kDefaultPedestrian_TransitStartEndMaxDistance = 2415; // 1.5 miles
+constexpr uint32_t kDefaultPedestrian_TransitTransferMaxDistance = 805;  // 0.5 miles
+
+// Bicycle defaults
+constexpr float kDefaultBicycle_ManeuverPenalty = 5.0f;        // Seconds
+constexpr float kDefaultBicycle_DrivewayPenalty = 300.0f;      // Seconds
+constexpr float kDefaultBicycle_AlleyPenalty = 60.0f;          // Seconds
+constexpr float kDefaultBicycle_GateCost = 30.0f;              // Seconds
+constexpr float kDefaultBicycle_GatePenalty = 300.0f;          // Seconds
+constexpr float kDefaultBicycle_FerryCost = 300.0f;            // Seconds
+constexpr float kDefaultBicycle_CountryCrossingCost = 600.0f;  // Seconds
+constexpr float kDefaultBicycle_CountryCrossingPenalty = 0.0f; // Seconds
+constexpr float kDefaultBicycle_UseRoad = 0.25f;               // Factor between 0 and 1
+constexpr float kDefaultBicycle_UseFerry = 0.5f;               // Factor between 0 and 1
+constexpr float kDefaultBicycle_UseHills = 0.25f;
+constexpr float kDefaultBicycle_AvoidBadSurfaces = 0.25f; // Factor between 0 and 1
+const std::string kDefaultBicycle_BicycleType = "Hybrid"; // Bicycle type
+constexpr float kDefaultBicycle_CyclingSpeed[] = {
+    25.0f, // Road bicycle: ~15.5 MPH
+    20.0f, // Cross bicycle: ~13 MPH
+    18.0f, // Hybrid or "city" bicycle: ~11.5 MPH
+    16.0f  // Mountain bicycle: ~10 MPH
+};
+
+// Truck defaults
+constexpr float kDefaultTruck_ManeuverPenalty = 5.0f;          // Seconds
+constexpr float kDefaultTruck_DestinationOnlyPenalty = 600.0f; // Seconds
+constexpr float kDefaultTruck_AlleyPenalty = 5.0f;             // Seconds
+constexpr float kDefaultTruck_GateCost = 30.0f;                // Seconds
+constexpr float kDefaultTruck_GatePenalty = 300.0f;            // Seconds
+constexpr float kDefaultTruck_TollBoothCost = 15.0f;           // Seconds
+constexpr float kDefaultTruck_TollBoothPenalty = 0.0f;         // Seconds
+constexpr float kDefaultTruck_CountryCrossingCost = 600.0f;    // Seconds
+constexpr float kDefaultTruck_CountryCrossingPenalty = 0.0f;   // Seconds
+constexpr float kDefaultTruck_LowClassPenalty = 30.0f;         // Seconds
+constexpr float kDefaultTruck_TruckWeight = 21.77f;            // Metric Tons (48,000 lbs)
+constexpr float kDefaultTruck_TruckAxleLoad = 9.07f;           // Metric Tons (20,000 lbs)
+constexpr float kDefaultTruck_TruckHeight = 4.11f;             // Meters (13 feet 6 inches)
+constexpr float kDefaultTruck_TruckWidth = 2.6f;               // Meters (102.36 inches)
+constexpr float kDefaultTruck_TruckLength = 21.64f;            // Meters (71 feet)
+
+// Transit defaults
+constexpr float kDefaultTransit_ModeFactor = 1.0f; // Favor this mode?
+constexpr float kDefaultTransit_TransferCost = 15.0f;
+constexpr float kDefaultTransit_TransferPenalty = 300.0f;
+constexpr float kDefaultTransit_UseBus = 0.3f;
+constexpr float kDefaultTransit_UseRail = 0.6f;
+constexpr float kDefaultTransit_UseTransfers = 0.3f;
```

Comment:
Could these constants be imported from source instead of redeclared here in the tests? Having these numbers in two places is an opportunity for code to drift.

Reply:
I would prefer not to have in both places - the code won't drift because the test will fail and force an update. The trade-off was to have the costing options tests in one place for some reuse of common items.